src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -309,6 +375,49 @@ public Jwt parse(String jwt) throws ExpiredJwtException, MalformedJwtException,
         }
     }
 
+    private void validateExpectedClaims(Header header, Claims claims) {
+        for (String expectedClaimName : expectedClaims.keySet()) {
+
+            Object expectedClaimValue = expectedClaims.get(expectedClaimName);
+            Object actualClaimValue = claims.get(expectedClaimName);
+
+            if (Claims.ISSUED_AT.equals(expectedClaimName)) {
+                expectedClaimValue = expectedClaims.getIssuedAt();
```

Comment:
I don't think we need these 3 separate if statements if the get(name,date) method is adjusted as mentioned above.